src/fluid/fluid.cpp

```diff
@@ -190,7 +206,10 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   // Just want constant primitive fields around to serve as
   // background if we are not evolving the fluid, don't need
   // to do the rest.
-  if (!active) return physics;
+  // TODO(BRR) logic gets very complicated accounting for this in situations where
+  // radiation is active but fluid isn't -- for example we want fluid prims from c2p to
+  // calculate opacities.
+  // if (!active) return physics;
```

Comment:
I feel medium that the comment should remain, since I think there was some thought put into being able to not allocate fluid cons variables for non-fluid runs. One could maybe work around this even for the 4D radiation source solve by not allowing that version of source solving if the fluid isn't active, but I think that extra software engineering work isn't well-motivated right now. I'll make an issue to think about this later, though I don't imagine many scenarios where we really care about this much memory in this case.

Reply:
#129

Reply:
Yeah I find it hard to believe that the memory footprint here will be important. But up to you.